pkg/agent/monitortool/latency_store.go

```diff
@@ -0,0 +1,298 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitortool
+
+import (
+	"errors"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/containernetworking/plugins/pkg/ip"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/util/k8s"
+
+	coreinformers "k8s.io/client-go/informers/core/v1"
+)
+
+// LatencyStore is a store for latency information of connections between nodes.
+type LatencyStore struct {
+	// Maybe we need to use small lock for the map
+	mutex sync.RWMutex
+
+	// isNetworkPolicyOnly is the flag to indicate if the Antrea Agent is running in network policy only mode.
+	isNetworkPolicyOnly bool
+	// The map of node name to node info, it will changed to gw0 watcher
+	nodeInformer coreinformers.NodeInformer
+	// The map of node ip to connection
+	connectionMap map[string][]*Connection
+	// The map of node ip to node name, it will be changed to gw0 watcher
+	nodeGW0Map map[string][]net.IP
+}
+
+// TODO1: use LRU cache to store the latency of the connection?
+type Connection struct {
```

Comment:
"connection" is a bit of an inaccurate term here. Maybe `NodeIPLatencyEntry`?

Reply:
Sounds good, I will change it.